modules/nw-running-dpdk-rootless-tap.adoc

```diff
@@ -0,0 +1,235 @@
+* You have installed the OpenShift CLI (`oc`).
+* You have installed the SR-IOV Network Operator.
+* You are logged in as a user with `cluster-admin` privileges.
+* Ensure that `setsebools container_use_devices=on` is set as root on all nodes.
++
+[NOTE]
+====
+Use the Machine Config Operator to set this SELinux boolean.
+====
+
+.Procedure
+
+. Create a file, such as `test-namespace.yaml`, with content like the following example:
++
+[source,yaml]
+----
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: test-namespace
+  labels:
+    pod-security.kubernetes.io/enforce: privileged
+    pod-security.kubernetes.io/audit: privileged
+    pod-security.kubernetes.io/warn: privileged
+    security.openshift.io/scc.podSecurityLabelSync: "false"
+----
+
+. Apply the yaml by running the following command:
++
+[source,terminal]
+----
+$ oc apply -f test-namespace.yaml
+----
+
+. Create the following `SriovNetworkNodePolicy` object, and then save the YAML in the `sriov-node-network-policy.yaml` file:
++
+[source,yaml]
+----
+apiVersion: sriovnetwork.openshift.io/v1
+kind: SriovNetworkNodePolicy
+metadata:
+ name: sriovnic
+ namespace: openshift-sriov-network-operator
+spec:
+ deviceType: netdevice <1>
+ isRdma: true <2>
+ needVhostNet: true <3>
+ nicSelector:
+   vendor: "15b3" <4>
+   deviceID: "101b" <5>
+   rootDevices: ["00:05.0"]
+ numVfs: 10
+ priority: 99
+ resourceName: sriovnic
+ nodeSelector:
+    feature.node.kubernetes.io/network-sriov.capable: "true"
+----
+<1> This indicates that the profile is tailored specifically for Mellanox Network Interface Controllers (NICs).
+<2> Setting `isRdma` to `true` is only required for a Mellanox NIC.
+<3> This mounts the `/dev/net/tun` and `/dev/vhost-net` devices into the container so the application can create a tap device and connect the tap device to the DPDK workload.
+<4> The vendor hexadecimal code of the SR-IOV network device. The value 15b3 is associated with a Mellanox NIC.
+<5> The device hexadecimal code of the SR-IOV network device.
+
+. Create the `SriovNetworkNodePolicy` object by running the following command:
++
+[source,terminal]
+----
+$ oc create -f sriov-node-network-policy.yaml
+----
+
+. Create the following `SriovNetwork` object, and then save the YAML in the `sriov-network-attachment.yaml` file:
++
+[source,yaml]
+----
+apiVersion: sriovnetwork.openshift.io/v1
+kind: SriovNetwork
+metadata:
+ name: sriov-network
+ namespace: openshift-sriov-network-operator
+spec:
+ networkNamespace: test-namespace
+ resourceName: sriovnic
+ spoofChk: "off"
+ trust: "on"
+----
++
+[NOTE]
+=====
+See the "Configuring SR-IOV additional network" section for a detailed explanation on each option in `SriovNetwork`.
+=====
++
+An optional library, `app-netutil`, provides several API methods for gathering network information about a container's parent pod.
+
+. Create the `SriovNetwork` object by running the following command:
++
+[source,terminal]
+----
+$ oc create -f sriov-network-attachment.yaml
+----
```

Comment:
Why is this one `oc create` and the others are `oc apply`? Is there a difference?

Reply:
My understanding is there is little difference, but I am going to leave as is because this was the procedure verified.

Reply:
apply will also work. Let's change it to be consistent